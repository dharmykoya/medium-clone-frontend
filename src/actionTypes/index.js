export const APP_THEME = 'APP_THEME';

export const USER_SIGNUP_START = 'USER_SIGNUP_START';
export const USER_SIGNUP_SUCCESS = 'USER_SIGNUP_SUCCESS';
export const USER_SIGNUP_FAIL = 'USER_SIGNUP_FAIL';

export const LOG_IN_USER = 'LOG_IN_USER';
export const LOG_IN_USER_ERROR = 'LOG_IN_USER_ERROR';

export const SET_CURRENT_USER = 'SET_CURRENT_USER';

export const GET_SINGLE_ARTICLE_START = 'GET_SINGLE_ARTICLE_START';
export const GET_SINGLE_ARTICLE_SUCCESS = 'GET_SINGLE_ARTICLE_SUCCESS';
export const GET_SINGLE_ARTICLE_FAIL = 'GET_SINGLE_ARTICLE_FAIL';

export const CREATE_COMMENT_ON_ARTICLE_START =
  'CREATE_COMMENT_ON_ARTICLE_START';
export const CREATE_COMMENT_ON_ARTICLE_SUCCESS =
  'CREATE_COMMENT_ON_ARTICLE_SUCCESS';

export const ADD_ALL_ARTICLES = 'ADD_ALL_ARTICLES';
export const UPDATE_ARTICLES_LOADING = 'UPDATE_ARTICLES_LOADING';
export const ADD_ALL_TAGS = 'ADD_ALL_TAGS';
export const FETCH_ALL_ARTICLES = 'FETCH_ALL_ARTICLES';
export const FETCH_ALL_ARTICLES_START = 'FETCH_ALL_ARTICLES_START';
export const FETCH_MORE_ARTICLES = 'FETCH_MORE_ARTICLES';
export const UPDATE_EMPTY_NARROWED = 'UPDATE_EMPTY_NARROWED';
export const CREATE_ARTICLE_FAIL = 'CREATE_ARTICLE_FAIL';
export const CREATE_ARTICLE_SUCCESS = 'CREATE_ARTICLE_SUCCESS';
export const CREATE_ARTICLE_START = 'CREATE_ARTICLE_START';
export const CREATE_BOOKMARK_SUCCESS = 'CREATE_BOOKMARK_SUCCESS';
export const CREATE_BOOKMARK_FAIL = 'CREATE_BOOKMARK_FAIL';
export const GET_PUBLISHED_ARTICLES_START = 'GET_PUBLISHED_ARTICLES_START';
export const GET_PUBLISHED_ARTICLES_SUCCESS = 'GET_PUBLISHED_ARTICLES_SUCCESS';
export const GET_DRAFT_ARTICLES_START = 'GET_DRAFT_ARTICLES_START';
export const GET_DRAFT_ARTICLES_SUCCESS = 'GET_DRAFT_ARTICLES_SUCCESS';
export const UNPUBLISHED_ARTICLES_SUCCESS = 'UNPUBLISHED_ARTICLES_SUCCESS';
export const SINGLE_ARTILCE_SLUG = 'SINGLE_ARTILCE_SLUG';
export const ERROR_RESPONSE_ON_ARTICLE = 'ERROR_RESPONSE_ON_ARTICLE';
export const PUBLISHED_ARTICLES_SUCCESS = 'PUBLISHED_ARTICLES_SUCCESS';
export const GET_BOOKMARKED_ARTICLES_START = 'GET_BOOKMARKED_ARTICLES_START';
export const GET_BOOKMARKED_ARTICLES_SUCCESS =
  'GET_BOOKMARKED_ARTICLES_SUCCESS';
export const REMOVE_BOOKMARK_SUCCESS = 'REMOVE_BOOKMARK_SUCCESS';
export const RATE_ARTICLE = 'RATE_ARTICLE';
export const RATE_ARTICLE_ERROR = 'RATE_ARTICLE_ERROR';
export const CLEAN_UP_RATING = 'CLEAN_UP_RATING';

export const LIKE_COMMENT_ON_ARTICLE_SUCCESS =
  'LIKE_COMMENT_ON_ARTICLE_SUCCESS';
export const GET_COMMENT_ON_ARTICLE_SUCCESS = 'GET_COMMENT_ON_ARTICLE_SUCCESS';
export const GET_COMMENT_ON_ARTICLE_START = 'GET_COMMENT_ON_ARTICLE_START';
export const OPEN_NAVBAR = 'OPEN_NAVBAR';
export const CLOSE_NAVBAR = 'CLOSE_NAVBAR';
export const EDIT_PROFILE_START = 'EDIT_PROFILE_START';
export const EDIT_PROFILE_SUCCESS = 'EDIT_PROFILE_SUCCESS';
export const EDIT_PROFILE_FAIL = 'EDIT_PROFILE_FAIL';
export const FOLLOW_AUTHOR = 'FOLLOW_AUTHOR';
export const GET_USER_FOLLOWERS = 'GET_USER_FOLLOWERS';
export const CLEAN_UP_FOLLOW = 'CLEAN_UP_FOLLOW';
export const REQUEST_PASSWORD_RESET = 'REQUEST_PASSWORD_RESET';

export const GET_NOTIFICATION_SUCCESS = 'GET_NOTIFICATION_SUCCESS';
export const GET_NOTIFICATION_START = 'GET_NOTIFICATION_START';
export const UPDATE_COMMENT_START = 'UPDATE_COMMENT_START';
export const UPDATE_COMMENT = 'UPDATE_COMMENT';
export const UPDATE_COMMENT_FAIL = 'UPDATE_COMMENT_FAIL';
export const GET_COMMENT_EDIT_HISTORY_START = 'GET_COMMENT_EDIT_HISTORY_START';
export const GET_COMMENT_EDIT_HISTORY = 'GET_COMMENT_EDIT_HISTORY';
export const CLEAN_UP_COMMENT_EDIT_HISTORY = 'CLEAN_UP_COMMENT_EDIT_HISTORY';
