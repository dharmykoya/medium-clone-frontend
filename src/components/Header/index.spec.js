import React from 'react';
import { mount } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';
import { Header } from './index';
import * as actionTypes from '../../actionTypes';
import { themeToggler } from './index.action';

const defaultProps = {
  themeToggler: jest.fn(),
  theme: {
    theme: 'light-theme'
  },
  auth: {
    isAuthenticated: false,
    user: {
      image: '',
      firstName: ''
    }
  },
  getAllNotifications: jest.fn(),
  Notifications: {
    notifications: [{}]
  }
};

const actionArgs = {
  theme: 'light-theme'
};

describe('Render component', () => {
  it('should create an action ', () => {
    //arrange
    const expectedAction = {
      type: actionTypes.APP_THEME,
      payload: actionArgs
    };
    // act
    const action = themeToggler(actionArgs);
    //assert
    expect(action).toEqual(expectedAction);
  });

  it('should render component successfully', () => {
    const component = mount(
      <BrowserRouter>
        <Header {...defaultProps} />
      </BrowserRouter>
    );
    expect(
      component.contains(<h3 className="m-0">Author's Haven</h3>)
    ).toBeTruthy();
  });

  it('should toggle to light theme', () => {
    const component = mount(
      <BrowserRouter>
        <Header {...defaultProps} />
      </BrowserRouter>
    );
    component.find('.toggle').simulate('click');
    expect(component.find('.toggle').length).toEqual(1);
    expect(component).toMatchSnapshot();
  });

  it('should toggle to dark theme', () => {
    const component = mount(
      <BrowserRouter>
        <Header {...defaultProps} />
      </BrowserRouter>
    );
    component.find('.toggle').simulate('click');
    expect(component.find('.toggle .switch')).toBeTruthy();
    expect(component).toMatchSnapshot();
  });

  it('should toggle to dark theme', () => {
    const component = mount(
      <BrowserRouter>
        <Header
          {...{
            themeToggler: jest.fn(),
            theme: {
              theme: 'dark-theme'
            },
            auth: {
              isAuthenticated: false,
              user: {
                image: '',
                firstName: ''
              }
            },
            getAllNotifications: jest.fn(),
            Notifications: {
              notifications: [{}]
            }
          }}
        />
      </BrowserRouter>
    );
    component.find('.toggle').simulate('click');
    expect(component.find('.toggle .switch')).toBeTruthy();
    expect(component).toMatchSnapshot();
  });

  it('should check if user is authenticated', () => {
    const component = mount(
      <BrowserRouter>
        <Header
          {...{
            themeToggler: jest.fn(),
            theme: {
              theme: 'dark-theme'
            },
            auth: {
              isAuthenticated: true,
              user: {
                image: '',
                firstName: ''
              }
            },

            getAllNotifications: jest.fn(),
            Notifications: {
              notifications: [{}]
            }
          }}
        />
      </BrowserRouter>
    );
    component
      .find('.compose-btn')
      .at(0)
      .simulate('click');
    expect(component).toMatchSnapshot();
  });

  it('should logout user successfully', () => {
    const component = mount(
      <BrowserRouter>
        <Header
          {...{
            themeToggler: jest.fn(),
            logout: jest.fn(),
            theme: {
              theme: 'dark-theme'
            },
            auth: {
              isAuthenticated: true,
              user: {
                image: 'sam.jpg',
                firstName: ''
              }
            },

            getAllNotifications: jest.fn(),
            Notifications: {
              notifications: [{}]
            }
          }}
        />
      </BrowserRouter>
    );
    component
      .find('.logout')
      .at(0)
      .simulate('click');
    expect(component).toMatchSnapshot();
  });

  it('should toggle navbar correctly', () => {
    const component = mount(
      <BrowserRouter>
        <Header
          {...{
            themeToggler: jest.fn(),
            logout: jest.fn(),
            theme: {
              theme: 'dark-theme'
            },
            auth: {
              isAuthenticated: true,
              user: {
                image: 'sam.jpg',
                firstName: ''
              }
            },
            getAllNotifications: jest.fn(),
            Notifications: {
              notifications: [{}]
            }
          }}
        />
      </BrowserRouter>
    );
    component
      .find('button')
      .at(0)
      .simulate('click');
    expect(component).toMatchSnapshot();
  });
});
