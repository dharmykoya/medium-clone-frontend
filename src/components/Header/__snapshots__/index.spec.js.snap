// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render component should check if user is authenticated 1`] = `ReactWrapper {}`;

exports[`Render component should logout user successfully 1`] = `ReactWrapper {}`;

exports[`Render component should toggle navbar correctly 1`] = `ReactWrapper {}`;

exports[`Render component should toggle to dark theme 1`] = `ReactWrapper {}`;

exports[`Render component should toggle to dark theme 2`] = `ReactWrapper {}`;

exports[`Render component should toggle to light theme 1`] = `ReactWrapper {}`;
